crates/cargo-contract/src/cmd/extrinsics/mod.rs

```diff
@@ -238,6 +320,47 @@ fn error_details(error: &DispatchError, metadata: &subxt::Metadata) -> Result<St
     }
 }
 
+#[derive(serde::Serialize)]
+pub enum ErrorVariant {
+    #[serde(rename = "module_error")]
+    Module(ModuleError),
+    #[serde(rename = "generic_error")]
+    Generic(GenericError),
+}
+
+#[derive(serde::Serialize)]
+pub struct ModuleError {
+    pub pallet: String,
+    pub error: String,
+    pub docs: Vec<String>,
+}
+
+#[derive(serde::Serialize)]
+pub struct GenericError {
+    error: String,
+}
+
+fn error_details_object(
```

Comment:
This should be a constructor on `ErrorVariant`, either `new` or somenthing like `from_dispatch_error`. Then, you could even get rid of `fn error_details` and use this instead. Just need to implement `Display` on `ErrorVariant` to give a string instead of json.